src/v/cloud_storage/tests/cache_bench.cc

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_storage/access_time_tracker.h"
+#include "seastarx.h"
+#include "ssx/sformat.h"
+
+#include <seastar/testing/perf_tests.hh>
+
+static std::chrono::system_clock::time_point make_ts(int64_t val) {
+    auto seconds = std::chrono::seconds(val);
+    return std::chrono::system_clock::time_point(seconds);
+}
+
+static void run_test(int test_scale) {
+    cloud_storage::access_time_tracker tracker;
+    std::vector<ss::sstring> names;
+    for (int i = 0; i < test_scale; i++) {
+        names.push_back(ssx::sformat("name-i", i));
+    }
+
+    for (int i = 0; i < test_scale; i++) {
+        perf_tests::start_measuring_time();
+        tracker.add_timestamp(names[i % test_scale], make_ts(i));
+        perf_tests::stop_measuring_time();
```

Comment:
You could also just time the whole loop.. less time getting system clock. (just a comment--not requiring a change here)